update-translations.py

```diff
@@ -208,8 +210,61 @@ def postprocess_translations(reduce_diff_hacks=False):
             tree.write(filepath, encoding='utf-8')
     return have_errors
 
+def update_git():
+    '''
+    Add new files to git repository.
+    (Removing files isn't necessary here, as `git commit -a` will take care of removing files that are gone)
+    '''
+    file_paths = [filepath for (filename, filepath) in all_ts_files()]
+    subprocess.check_call([GIT, 'add'] + file_paths)
+
+def update_build_systems():
+    '''
+    Update build system and Qt resource descriptors.
+    '''
+    filename_lang = [re.match(r'((bitcoin_(.*)).ts)$', filename).groups() for (filename, filepath) in all_ts_files()]
+    filename_lang.sort(key=lambda x: x[0])
+
+    # update qrc locales
+    with open('src/qt/bitcoin_locale.qrc', 'w') as f:
+        f.write('<!DOCTYPE RCC><RCC version="1.0">\n')
+        f.write('    <qresource prefix="/translations">\n')
+        for (filename, basename, lang) in filename_lang:
+            f.write(f'        <file alias="{lang}">locale/{basename}.qm</file>\n')
+        f.write('    </qresource>\n')
+        f.write('</RCC>\n')
+
+    # update Makefile
+    with open('src/Makefile.qt.include', 'r') as f:
+        lines = list(f)
+    with open('src/Makefile.qt.include', 'w') as f:
```

Comment:
This part would be much simpler if the file with locale definitions was a separate include file, which could be fully generated instead of an existing file. This works right now, though.

Reply:
Did this in a new commit. I retain the old code in the older commit for older versions, though it'd likely be better to backport bitcoin/bitcoin#16982 to all releases that we care about.